File: android/app/src/main/java/com/github/makewheels/video2022/data/model/Channel.kt
package com.github.makewheels.video2022.data.model

data class ChannelInfo(
    val userId: String,
    val nickname: String?,
    val avatarUrl: String?,
    val bannerUrl: String?,
    val bio: String?,
    val subscriberCount: Long = 0,
    val videoCount: Long = 0,
    val isSubscribed: Boolean = false
)

data class UpdateProfileRequest(
    val nickname: String?,
    val bio: String?
)

File: ios/VideoApp/Sources/Models/Channel.swift
import Foundation

struct ChannelInfo: Decodable {
    let userId: String
    let nickname: String?
    let avatarUrl: String?
    let bannerUrl: String?
    let bio: String?
    let subscriberCount: Int
    let videoCount: Int
    var isSubscribed: Bool
}

struct UpdateProfileRequest: Encodable {
    let nickname: String?
    let bio: String?
}
